lib/interfaces/config-module-options.interface.ts

```diff
@@ -59,6 +59,12 @@ export interface ConfigModuleOptions {
    */
   load?: Array<ConfigFactory>;
```

Comment:
can't we use the same `load` to that instead of introducing yet another option?

```diff
- load?: Array<ConfigFactory>;
+ load?: Array<ConfigFactory | Promise<ConfigFactory>>;
```

I guess

Reply:
I introduced the additional option to clearly express the intention behind the change. However, I acknowledge your point that utilizing the existing 'load' option to accommodate asynchronous 'ConfigFactory' objects would be a more efficient approach. I will proceed with modifying the existing 'load' to support asynchronous operations. Thank you for bringing this to my attention.